config/confighttp/compress_readcloser.go

```diff
@@ -0,0 +1,23 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package confighttp // import "go.opentelemetry.io/collector/config/confighttp"
+
+import "io"
+
+// compressReadCloser couples the original compressed reader
+// and the compression reader to ensure that the original body
+// is correctly closed to ensure resources are freed.
+type compressReadCloser struct {
```

Comment:
Why not just `readCloser`?

Reply:
I mostly wanted to make it clear that the usage is intended for compression.